test/vanilla/Expected/AcceptanceTests/ExtensibleEnums/Models/DaysOfWeekExtensibleEnum.cs

```diff
@@ -0,0 +1,65 @@
+// <auto-generated>
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+//
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+// </auto-generated>
+
+namespace Fixtures.AcceptanceTestsExtensibleEnums.Models
+{
+    using Microsoft.Rest;
+    using Microsoft.Rest.ClientRuntime;
+    using Microsoft.Rest.ClientRuntime.Serialization;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Defines values for DaysOfWeekExtensibleEnum.
+    /// </summary>
+    [JsonConverter(typeof(Microsoft.Rest.ClientRuntime.Serialization.ExtensibleEnumConverter<DaysOfWeekExtensibleEnum, string>))]
+    public sealed class DaysOfWeekExtensibleEnum : Microsoft.Rest.ExtensibleEnum<DaysOfWeekExtensibleEnum, string>
+    {
+        private DaysOfWeekExtensibleEnum(string value):base(value)
+        {
+        }
+
+        public static readonly DaysOfWeekExtensibleEnum Monday = "Monday";
+
+        public static readonly DaysOfWeekExtensibleEnum Tuesday = "Tuesday";
+
+        public static readonly DaysOfWeekExtensibleEnum Wednesday = "Wednesday";
+
+        public static readonly DaysOfWeekExtensibleEnum Thursday = "Thursday";
+
+        public static readonly DaysOfWeekExtensibleEnum Friday = "Friday";
+
+        public static readonly DaysOfWeekExtensibleEnum Saturday = "Saturday";
+
+        public static readonly DaysOfWeekExtensibleEnum Sunday = "Sunday";
+
+
+        /// <summary>
+        /// Defines ctor/explicit conversion from value type to
+        /// DaysOfWeekExtensibleEnum.
+        /// </summary>
+        /// <param name="value">string to convert.</param>
+        public static implicit operator DaysOfWeekExtensibleEnum(string value) => Create(value);
+
+        /// <summary>
+        /// Create method used by deserializer to create instances of DaysOfWeekExtensibleEnum type
+        /// </summary>
+        /// <returns>The value as a DaysOfWeekExtensibleEnum.</returns>
+        public static DaysOfWeekExtensibleEnum Create(string value)
+        {
+            if(value == null)
+            {
+                return null;
```

Comment:
why?

Reply:
Would it make sense to generate an ExtensibleEnum object for a null underlying value?

Reply:
Because `null ` indicates the absence of a value, not a distinct value.

If `null` is a possible value, then the field isn't required.

Reply:
Yup, `x-nullable` would be the right thing to do

Reply:
Right, so having this check essentially makes the assumption that stuff is always nullable (whether via `required: false` or `x-nullable: true`).
Furthermore, when `null` actually *is* one of the enum values (perfectly valid Swagger), I'd somewhat expect it to end up in an `ExtensibleEnum` instance. In the *absence* of a value on the other hand, why would one even call `Create`? This check essentially shuts the door on an (arguably not common so far?) use case *but* at not visible (to me) benefit.

Reply:
I'm with @olydis on this one. `ExtensibleEnum` is a class, so it is already possible to have a `null` instance. Why allow `null` for `value` at all?